libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/sound/MapboxAudioComponentState.kt

```diff
@@ -0,0 +1,19 @@
+package com.mapbox.navigation.dropin.component.sound
+
+import com.mapbox.api.directions.v5.models.VoiceInstructions
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.ui.voice.model.SpeechAnnouncement
+
+/**
+ * @param isPlayable true when the active route has voice instructions
+ * @param isMuted user controlled value to mute or un-mute audio guidance
+ * @param voiceInstructions the current voice instructions
+ * @param speechAnnouncement the current speech announcements
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+data class MapboxAudioComponentState internal constructor(
+    val isPlayable: Boolean = false,
```

Comment:
I am thinking instead of introducing these additional flags and doing the same for other views where ever applicable, why not use the `NavigationState` and keep it simple. You are using this boolean flag to decide the visibility. Why not just use `NavigationState` and make it visible only in ActiveGuidance while hide in all other screens?

Reply:
The way we implemented it in 1tap, is always play voice instructions whenever they come. That way, it is navigation core that decides when to play voice instructions. The user, is able to mute it with a button.

If we depend on NavigationState, we will show the audio button when there is no audio messages available; or we will not show the audio button when audio is playing.

Reply:
Okay this also makes sense. Sounds good to me

Reply:
Sorry back and forth here. Other change suggested offline is having an action list per `NavigationState`. This feel like a good idea. Also is simpler to not have the visibility flow.

There may be some complications with Android Auto, but I'm imagining some solutions that will work well with it. Android Auto will need access the `NavigationState` to ensure audio guidance will play for both. 

Updated